imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -121,18 +200,27 @@ private void parseDetailRecord(CallDetailRecordDto cdr, Map<String, CallSummaryR
     }
 
 
-    // Create an in-memory set of summary records from all detail records
-    private ParseResults parseDetailRecords(File file, FileInfo fileInfo) {
+    private int getMaxErrorCount() {
+        try {
+            return Integer.parseInt(settingsFacade.getProperty(MAX_CDR_ERROR_COUNT));
+        } catch (NumberFormatException e) {
+            return MAX_CDR_ERROR_COUNT_DEFAULT;
+        }
+    }
+
+
+    @Override
+    public VerifyResults aggregateDetailFile(File file, FileInfo fileInfo, boolean verifyOnly) {
```

Comment:
Nit pick on naming.  This method is called 'aggregateDetailFile' but it returns a 'VerifyResults' object.  It also can be called such that it doesn't aggregate.

Maybe change the name to aggregateAndVerifyDetailFile?